### ISSUE TITLE
Mutations seem to keep repeating after sucessfull DELETE

### ISSUE THREAD

**Issue by WouldYouKindly** [yandex/ClickHouse#2961]
Hi.

We have a table with structure like this:

```
name───────────────────type───
r_id                    Int32  
t_date                  Date   
t_start_minute          Int32  
t_end_minute            Int32  
c_id                    Int32  
h_id                    Int32  
size                    Float64
factor                  Float64
weight                  Float64
age                     Int8   
param1                  UInt8  
param2                  Int8   
param3                  Int8   
param4                  Int8   
param5                  Int8   
```

`ENGINE = MergeTree(t_date, (t_date, t_start_minute, r_id), 8192);`

We perform a bunch of DELETEs with `command` = `DELETE WHERE t_date = '2014-08-02'`.
Most of them have `is_done = 1` after a couple of seconds. But some haven't.

Data are deleted from the table. But Clickhouse keeps repeating the following log messages:

```
<Error> void DB::BackgroundProcessingPool::threadFunction(): Code: 10, e.displayText() = DB::Exception: Not found column t_end_minute in block. There are only columns: , e.what() = DB::Exception, Stack trace:

0. /usr/bin/clickhouse-server(StackTrace::StackTrace()+0x16) [0x56d1436]
1. /usr/bin/clickhouse-server(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x22) [0x2db80f2]
2. /usr/bin/clickhouse-server(DB::Block::getByName(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0xe7) [0x4755f07]
3. /usr/bin/clickhouse-server(DB::ColumnGathererStream::init()+0x1dd) [0x51af9ad]
4. /usr/bin/clickhouse-server(DB::ColumnGathererStream::readImpl()+0x8e0) [0x51b1830]
5. /usr/bin/clickhouse-server(DB::IProfilingBlockInputStream::read()+0x25a) [0x4768cfa]
6. /usr/bin/clickhouse-server(DB::MergeTreeDataMergerMutator::mergePartsToTemporaryPart(DB::MergeTreeDataMergerMutator::FuturePart const&, DB::MergeListEntry&, unsigned long, long, DB::DiskSpaceMonitor::Reservation*, bool)+0x3c97) [0x50631b7]
7. /usr/bin/clickhouse-server(DB::StorageMergeTree::merge(unsigned long, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)+0x50f) [0x4fb5faf]
8. /usr/bin/clickhouse-server(DB::StorageMergeTree::backgroundTask()+0x174) [0x4fb8454]
9. /usr/bin/clickhouse-server(DB::BackgroundProcessingPool::threadFunction()+0x47e) [0x501da9e]
10. /usr/bin/clickhouse-server() [0x938cc0f]
11. /lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba) [0x7fd8b0a986ba]
12. /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d) [0x7fd8b02b041d]
```

Clickhouse version: 18.6.0.

**Comment by ztlpn** [comment]
The bug (vertical merge fails for empty parts that are created as a result of DELETE) was fixed in #2930.

To fix the problem before this fix is released, you can temporarily turn off vertical merge (set `enable_vertical_merge_algorithm` to 0 in [MergeTree settings](https://clickhouse.yandex/docs/en/operations/server_settings/settings/#merge_tree)). It should be turned off until there are no empty parts left (`SELECT count() FROM system.parts WHERE rows = 0 AND active` returns 0). Don't forget to turn it on later!

**Comment by WouldYouKindly** [comment]
@ztlpn Thank you! We found that in testing, so we just dropped the table (: 
But good to know that it's fixed.
